edb/pgsql/compiler/dml.py

```diff
@@ -459,80 +529,11 @@ def process_insert_body(
-        pseudo_iterator_cte = None
-        pseudo_iterator_set = None
-
-    iterator_cte: Optional[pgast.CommonTableExpr]
-    iterator_id: Optional[pgast.BaseExpr]
-
-    if iterator_set is not None:
-        with ctx.substmt() as ictx:
-            # This has been lifted up to the top level, so we can't
-            # have it inheriting anything from where it would have
-            # been nested.
-            # TODO: Inherit from the nearest iterator instead?
-            del ictx.rel_hierarchy[ictx.rel]
-            ictx.path_scope = ictx.path_scope.new_child()
-            ictx.path_scope[iterator_set.path_id] = ictx.rel
-
-            # If there is an enclosing insert *and* an explicit iterator,
-            # we need to correlate the enclosing insert with the iterator.
-            if pseudo_iterator_set is not None:
-                assert pseudo_iterator_cte
-                pseudo_iterator_rvar = relctx.rvar_for_rel(
-                    pseudo_iterator_cte, ctx=ictx)
-
-                pathctx.put_path_bond(ictx.rel, pseudo_iterator_set.path_id)
-                relctx.include_rvar(
-                    ictx.rel, pseudo_iterator_rvar,
-                    path_id=pseudo_iterator_set.path_id,
-                    overwrite_path_rvar=True,
-                    ctx=ictx)
-
-                ictx.volatility_ref = pathctx.get_path_identity_var(
-                    ictx.rel,
-                    pseudo_iterator_set.path_id,
-                    env=ictx.env)
-
-            clauses.compile_iterator_expr(ictx.rel, iterator_set, ctx=ictx)
-            ictx.rel.path_id = iterator_set.path_id
-            pathctx.put_path_bond(ictx.rel, iterator_set.path_id)
-            iterator_cte = pgast.CommonTableExpr(
-                query=ictx.rel,
-                name=ctx.env.aliases.get('iter')
-            )
-            ictx.toplevel_stmt.ctes.append(iterator_cte)
-        iterator_rvar = relctx.rvar_for_rel(iterator_cte, ctx=ctx)
-        relctx.include_rvar(select, iterator_rvar,
-                            path_id=ictx.rel.path_id, ctx=ctx)
-        iterator_id = pathctx.get_path_identity_var(
-            select, iterator_set.path_id, env=ctx.env)
-
-    elif pseudo_iterator_set is not None:
-        assert pseudo_iterator_cte is not None
-        iterator_rvar = relctx.rvar_for_rel(pseudo_iterator_cte, ctx=ctx)
-        relctx.include_rvar(select, iterator_rvar,
-                            path_id=pseudo_iterator_set.path_id, ctx=ctx)
-
-        iterator_set = pseudo_iterator_set
-        iterator_cte = pseudo_iterator_cte
-        iterator_id = relctx.get_path_var(
-            select, pseudo_iterator_set.path_id,
-            aspect='identity', ctx=ctx)
-
-    else:
-        iterator_cte = None
-        iterator_id = None
+    iterator_id: Optional[pgast.BaseExpr] = None
+    iterator = ctx.enclosing_cte_iterator
+    if iterator is not None:
+        # XXX: THAT PATH BOND????
+        pathctx.put_path_bond(insert_stmt, iterator.path_id)
```

Comment:
You do this below, is it needed earlier?

Reply:
No